src/bun.js/webcore/streams.zig

```diff
@@ -3552,6 +3588,31 @@ pub const FileBlobLoader = struct {
             }
         }
 
+        if (this.poll_ref) |poll| {
+            const is_readable = poll.isReadable();
```

Comment:
@KProtty does this look right to you? For reading a blocking pipe without actually blocking

Reply:
Does this select/poll/epoll/kqueue on a pipe's readable fd, then do the read()? If so, then barring any concurrent reads done in between, it looks right at a glance.

Reply:
it does, thanks for looking